packages/dapp-kit-react-privy/src/DAppKitPrivyProvider/utils/AccountAbstraction.tsx

```diff
@@ -0,0 +1,255 @@
+import {
+    ABIFunction,
+    Address,
+    Clause,
+    Hex,
+    Secp256k1,
+} from '@vechain/sdk-core';
+import {
+    type ConnectedWallet,
+    type SignMessageModalUIOptions,
+    type SignTypedDataParams,
+} from '@privy-io/react-auth';
+import { type Abi, encodeFunctionData } from 'viem';
+import {
+    ProviderInternalBaseWallet,
+    signerUtils,
+    VeChainProvider,
+    type VeChainSigner,
+} from '@vechain/sdk-network';
+import { getAbstractAddress } from './AbstractAccount';
+import { DELEGATOR_URL, NETWORK_URL, THOR_CLIENT } from './Constants';
+
+/**
+ * Get the genesis block id
+ *
+ * @returns The genesis block id
+ */
+export async function getGenesisId(): Promise<string | null> {
+    const genesis = await THOR_CLIENT.blocks.getGenesisBlock();
+    return genesis?.id !== null && genesis?.id !== undefined
+        ? BigInt(genesis.id).toString()
+        : null;
+}
+
+/**
+ * Generate a random transaction user
+ * @returns The private key and address of the random transaction user
+ */
+export const randomTransactionUser = async (): Promise<{
+    privateKey: Buffer;
+    address: string;
+}> => {
+    const privateKey = Buffer.from(await Secp256k1.generatePrivateKey());
+    const address = Address.ofPrivateKey(Hex.of(privateKey).bytes);
+    return { privateKey, address: address.toString() };
+};
+
+/**
+ * Send transaction to the blockchain
+ * with the account abstraction function
+ */
+export async function sendTransactionFunction({
```

Comment:
It seems to me that we are duplicating a lot of code, we already have this in useSmartAccount